app/src/main/java/com/pavesid/androidacademy/repositories/remote/MoviesRemoteRepository.kt

```diff
@@ -0,0 +1,50 @@
+package com.pavesid.androidacademy.repositories.remote
+
+import com.pavesid.androidacademy.data.Actor
+import com.pavesid.androidacademy.data.Genre
+import com.pavesid.androidacademy.data.JsonActor
+import com.pavesid.androidacademy.data.JsonGenre
+import com.pavesid.androidacademy.data.JsonMovie
+import com.pavesid.androidacademy.data.Movie
+import com.pavesid.androidacademy.repositories.MoviesRepository
+import javax.inject.Inject
+
+class MoviesRemoteRepository @Inject constructor() : MoviesRepository {
+
+    override suspend fun getMovies(): List<Movie> = parseMovies(
+        MoviesApiImpl.getMovies(),
+        MoviesApiImpl.getGenres(),
+        MoviesApiImpl.getActors()
```

Comment:
In this case, your requests process consistently. You can try to run that's parallel by Flow (in one of the next pull-request I think)